clients/src/main/java/org/apache/kafka/common/serialization/StringSerializer.java

```diff
@@ -34,7 +34,7 @@ public void configure(Map<String, ?> configs, boolean isKey) {
         Object encodingValue = configs.get(propertyName);
         if (encodingValue == null)
             encodingValue = configs.get("serializer.encoding");
-        if (encodingValue != null && encodingValue instanceof String)
```

Comment:
`encodingValue` can still be null right, as `configs.get(..)` could still return a null.

Reply:
yes encoding value can be null but the instanceof operator does not return
null pointer exception for encodingValue==null.
Instead if encodingValue==null, (encodingValue instanceof String) returns
false which is the desired behavior.
Thanks for the quick reply.